vcd/resource_vcd_vm_internal_disk_test.go

```diff
@@ -0,0 +1,468 @@
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "storage_profile", storageProfile),
+					//resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "thin_provisioned", strconv.FormatBool(*adminVdc.IsThinProvision)),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName, "thin_provisioned", "true"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "size_in_mb", biggerDiskSize),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "iops", "0"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "allow_vm_reboot", "true"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "vcd_vm_internal_disk." + diskResourceName + "-import",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdVmObject(testConfig.VCD.Org, vdcName, vappName, vmName, "3000"),
+				// These fields can't be retrieved
+				ImportStateVerifyIgnore: []string{"org", "vdc", "allow_vm_reboot", "thin_provisioned"},
+			},
+			/*resource.TestStep{
+				Config: configText_update2,
+				Check: resource.ComposeTestCheckFunc(resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "size_in_mb", diskSize),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "bus_type", "ide"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "bus_number", "1"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "unit_number", "0"),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "storage_profile", testConfig.VCD.ProviderVdc.StorageProfile),
+					resource.TestCheckResourceAttr("vcd_vm_internal_disk."+diskResourceName+"_ide", "allow_vm_reboot", "true"),
+				),
+			},*/
+		},
+	})
+}
+
+func testCheckInternalDiskNonStringOutputs(internalDiskSize int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		outputs := s.RootModule().Outputs
+
+		if outputs["internal_disk_size"].Value != internalDiskSize {
+			return fmt.Errorf("internal disk size value didn't match")
+		}
+
+		if outputs["internal_disk_iops"].Value != 0 {
+			return fmt.Errorf("internal disk iops value didn't match")
+		}
+
+		if outputs["internal_disk_bus_type"].Value != "paravirtual" {
+			return fmt.Errorf("internal disk bus type value didn't match")
+		}
+
+		if outputs["internal_disk_bus_number"].Value != 0 {
+			return fmt.Errorf("internal disk bus number value didn't match")
+		}
+
+		if outputs["internal_disk_unit_number"].Value != 0 {
+			return fmt.Errorf("internal disk unit number value didn't match")
+		}
+
+		if outputs["internal_disk_storage_profile"].Value != "*" {
+			return fmt.Errorf("internal disk storage profile value didn't match")
+		}
+
+		return nil
+	}
+}
+
+/*func getAdminVdc() (*types.AdminVdc, error) {
```

Comment:
Is this still needed?

Reply:
will remove when I sure we finished changing design